DataFormats/GeometryVector/interface/PhiInterval.h

```diff
@@ -0,0 +1,42 @@
+#ifndef GeometryVector_PhiInterval_H
+#define GeometryVector_PhiInterval_H
+#include "DataFormats/GeometryVector/interface/Basic3DVector.h"
+#include "DataFormats/Math/interface/normalizedPhi.h"
+
+class PhiInterval {
+public:
+  PhiInterval(float phi1, float phi2) {
+    phi2 = proxim(phi2,phi1);
+    constexpr float c1 = 2.*M_PI;
+    if (phi2<phi1) phi2+=c1;
+    auto dphi = 0.5f*(phi2-phi1);
+    auto phi = phi1+dphi;
+    x = std::cos(phi);
+    y = std::sin(phi);
+    dcos = std::cos(dphi);
+  }
+
+  PhiInterval(float ix, float iy, float dphi) {
+    auto norm = 1.f/std::sqrt(ix*ix+iy*iy);
+    x = ix*norm;
+    y = iy*norm;
+    dcos = std::cos(dphi);
+  }
+
+  template<typename T>
+  bool inside(Basic3DVector<T> const & v) const {
+    return inside(v.x(),v.y());
+  }
+
+  bool inside(float ix, float iy) const {
```

Comment:
#include<cmath>
float x,y,dcos;
bool inside(float ix, float iy) {
    return ix*x+iy*y > dcos*std::sqrt(ix*ix+iy*iy);
  }

bool inside2(float ix, float iy) {
  auto norm = 1.f/std::sqrt(ix*ix+iy*iy);
    return norm*(ix*x+iy*y) > dcos;
  }

produces with Ofast
the real point is not the number of mul instruction, but how the out-of-order processor can feed the two FP engines and how the mul are pipelined : I do not think we can easily compute the latency of such a small kernel...

inside(float, float):                            # @inside(float, float)
        movss   x(%rip), %xmm2          # xmm2 = mem[0],zero,zero,zero
        mulss   %xmm0, %xmm2
        movss   y(%rip), %xmm3          # xmm3 = mem[0],zero,zero,zero
        mulss   %xmm1, %xmm3
        addss   %xmm2, %xmm3
        mulss   %xmm0, %xmm0
        mulss   %xmm1, %xmm1
        addss   %xmm0, %xmm1
        xorps   %xmm0, %xmm0
        rsqrtss %xmm1, %xmm0
        movaps  %xmm0, %xmm2
        mulss   %xmm2, %xmm2
        mulss   %xmm1, %xmm2
        addss   .LCPI2_0(%rip), %xmm2
        mulss   .LCPI2_1(%rip), %xmm0
        mulss   %xmm1, %xmm0
        mulss   %xmm2, %xmm0
        xorps   %xmm2, %xmm2
        cmpeqss %xmm2, %xmm1
        andnps  %xmm0, %xmm1
        mulss   dcos(%rip), %xmm1
        ucomiss %xmm1, %xmm3
        seta    %al
        retq

inside2(float, float):                           # @inside2(float, float)
        movaps  %xmm0, %xmm2
        mulss   %xmm2, %xmm2
        movaps  %xmm1, %xmm3
        mulss   %xmm3, %xmm3
        addss   %xmm2, %xmm3
        xorps   %xmm2, %xmm2
        rsqrtss %xmm3, %xmm2
        movaps  %xmm2, %xmm4
        mulss   %xmm4, %xmm4
        mulss   %xmm3, %xmm4
        addss   .LCPI3_0(%rip), %xmm4
        mulss   .LCPI3_1(%rip), %xmm2
        mulss   x(%rip), %xmm0
        mulss   y(%rip), %xmm1
        addss   %xmm0, %xmm1
        mulss   %xmm2, %xmm1
        mulss   %xmm4, %xmm1
        ucomiss dcos(%rip), %xmm1
        seta    %al